invenio_files_rest/tasks.py

```diff
@@ -264,3 +264,40 @@ def remove_expired_multipartobjects():
 
     for fid in file_ids:
         remove_file_data.delay(fid)
+
+
+@shared_task()
+def clear_orphaned_files(force=False):
+    """Delete orphaned files from DB and storage.
+
+    .. note:: Orphan files are files
+        (:class:`invenio_files_rest.models.FileInstance`s and their on-disk
+        counterparts) that do not have any
+        :class:`invenio_files_rest.models.ObjectVersion`s associated with them
+        (anymore).
+    """
+    # with the tilde (~) operator, we get all file instances that *don't*
+    # have any associated object versions
+    orphans = FileInstance.query.filter(~FileInstance.objects.any()).all()
+    for orphan in orphans():
+        file_loc = orphan.uri
```

Comment:
I would probably just send off a ``remove_file_data`` task:

``remove_file_data.delay(orphan.id)``

E.g. this ensure each files is removed in the most safe way. E.g. it will remove the file instance, commit to the database, and then remove the file. This way we're never left with a file instance in the database that does not have a corresponding file (except if someone from outside decided to remove it).

Reply:
**that** is also the task that inspired the CLI command!